docs/source/storage.rst

```diff
@@ -1,5 +1,122 @@
+bytes of the tensor.
+In general, a tensor created in PyTorch using regular constructors such as :func:`~torch.zeros`, :func:`~torch.zeros_like`
+or :func:`~torch.Tensor.new_zeros` will produce tensors where there is a one-to-one correspondence between the tensor
+storage and the tensor itself.
+
+But a storage also allows multiple tensors to share the same storage.
+For instance, any view of a tensor (obtained through :meth:`~torch.Tensor.view` or some but not all kinds of indexing
+like integers and slices) will point to the same underlying tensor as the original one.
+When serializing and deserializing tensors that share a common storage, the relationship is preserved, and the tensors
+continue to point to the same storage. Interestingly, deserializing multiple tensors that point to a single storage
+can be faster than deserializing multiple independent tensors.
+
+A tensor storage can be accessed through the :meth:`~torch.Tensor.untyped_storage` method. This will return an object of
+type :class:`torch.UntypedStorage`.
+Fortunately, storages have a unique identifier called accessed through the :meth:`torch.UntypedStorage.data_ptr` method.
+In regular settings, two tensors with the same data storage will have the same storage ``data_ptr``. However, this is
+not true for the tensors themselves: for instance, a tensor with a ``data`` and a ``grad`` value actually points to two
+independent storages and hence require a ``data_ptr()`` of its own.
+
+Untyped storages are somewhat independent of the tensors that are built on them. Practically, this means that tensors
+with different dtypes or shape can point to the same storage. It also implies that a tensor storage can be changed, as
+the following example shows:
+
+    >>> t = torch.ones(3)
+    >>> s0 = t.untyped_storage()
+    >>> s0
+     0
+     0
+     128
+     63
+     0
+     0
+     128
+     63
+     0
+     0
+     128
+     63
+    [torch.storage.UntypedStorage(device=cpu) of size 12]
+    >>> s1 = s0.clone()
+    >>> s1.fill_(0)
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+     0
+    [torch.storage.UntypedStorage(device=cpu) of size 12]
+    >>> # Fill the tensor with a zeroed storage
+    >>> t.set_(s1, storage_offset=t.storage_offset(), stride=t.stride(), size=t.size())
+    tensor([0., 0., 0.])
+
+Other than their ``data_ptr``, untyped storage also have other attributes such as :attr:`~torch.UntypedStorage.filename`
+(in case the storage points to a file on disk), :attr:`~torch.UntypedStorage.device` or
+:attr:`~torch.UntypedStorage.is_cuda` for device checks. A storage can also be manipulated in-place or
+out-of-place with methods like :attr:`~torch.UntypedStorage.copy_`, :attr:`~torch.UntypedStorage.fill_` or
+:attr:`~torch.UntypedStorage.pin_memory`. FOr more information, check the API
+reference below. Keep in mind that modifying storages is a low-level API and comes with some risks.
```

Comment:
```suggestion
reference below. Keep in mind that modifying storages is a low-level API and comes with risks!
```

Also mention that those same APIs are on the highlevel Tensor objects and should be what people try first?